### PR TITLE
Refactor photo task fragment

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Things done:
 * Removes the delete button as "undo" button is already present for the same purpose
 * Remove usage of rx based code and replace with coroutine
 * Remove obsolete / dead code 
 * Simplify logic by removing unnecessary indirections

[photo_capture.webm](https://github.com/google/ground-android/assets/8918023/7d400ba5-4e61-4338-8567-edfa8c5400c6)

@gino-m  PTAL?

### CODE REVIEW

File: ground/src/main/res/layout/photo_task_frag.xml [hunk 1]
```diff
@@ -32,19 +35,19 @@
...
       style="@style/TaskChipButton.LightGreen"
       android:layout_width="wrap_content"
       android:layout_height="wrap_content"
-      app:icon="@drawable/outline_photo_camera"
-      app:iconTint="?attr/colorOnSurface"
-      android:onClick="@{()-> viewModel.onTakePhotoClick()}"
+      android:onClick="@{()-> fragment.onTakePhoto()}"
       android:text="@string/take_picture"
-      android:visibility="@{viewModel.isPhotoPresent() ? View.GONE : View.VISIBLE}" />
+      android:visibility="@{viewModel.isPhotoPresent() ? View.GONE : View.VISIBLE}"
```
Comment (gino-m) [inline ground/src/main/res/layout/photo_task_frag.xml:48]: We have a binder for this, so you can now use app:visible with `@{viewModel.isPhotoPresent()}`

**Review by gino-m — CHANGES_REQUESTED** [review]
LG! A few small suggestions.

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/photo/PhotoResult.kt [hunk 2]
```diff
@@ -18,13 +18,11 @@ package com.google.android.ground.ui.datacollection.tasks.photo
...
-  val bitmap: Bitmap? = null,
-  val path: String? = null,
-  val isHandled: Boolean = false
-) {
-  fun isEmpty(): Boolean = bitmap == null && path == null
+data class PhotoResult constructor(val taskId: String, val bitmap: Bitmap?, val path: String?) {
+
+  init {
+    check(bitmap != null || !path.isNullOrEmpty()) {
+      "At-least one of bitmap or path should be non-null, found bitmap=$bitmap, path=$path"
```
[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/photo/PhotoResult.kt]: Done

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/photo/PhotoResult.kt [hunk 3]
```diff
@@ -16,13 +16,16 @@
...
+import com.google.android.ground.util.isNotNullOrEmpty
 
 /** Contains the bitmap or path to the photo a user captured or selected. */
 data class PhotoResult constructor(val taskId: String, val bitmap: Bitmap?, val path: String?) {
 
   init {
-    check(bitmap != null || !path.isNullOrEmpty()) {
-      "At-least one of bitmap or path should be non-null, found bitmap=$bitmap, path=$path"
+    check(bitmap != null || path.isNotNullOrEmpty()) {
+      "At-least one of bitmap or path should be non-null, found " +
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/photo/PhotoResult.kt:26]: Nit: At-least -> At least

File: ground/src/main/res/layout/photo_task_frag.xml [hunk 4]
```diff
@@ -41,11 +40,11 @@
     <com.google.android.material.button.MaterialButton
       android:id="@+id/btn_take_photo"
       style="@style/TaskChipButton.LightGreen"
+      visible="@{!viewModel.isPhotoPresent()"
```
Comment (gino-m) [inline ground/src/main/res/layout/photo_task_frag.xml:43]: Missing final `}`?

File: ground/src/main/java/com/google/android/ground/util/StringExt.kt [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.util
+
+fun String?.isNotNullOrEmpty(): Boolean = !this.isNullOrEmpty()
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/util/StringExt.kt:18]: Sorry I thought this existed already. If not I don't know if it's worth adding as an ext function, your call.

**Review by gino-m — CHANGES_REQUESTED** [review]

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/photo/PhotoResult.kt (hunk 3 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/photo/PhotoResult.kt]: Done

File: ground/src/main/res/layout/photo_task_frag.xml (hunk 4 above)
Comment (shobhitagarwal1612) [reply ground/src/main/res/layout/photo_task_frag.xml]: Done

File: ground/src/main/java/com/google/android/ground/util/StringExt.kt (hunk 5 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/util/StringExt.kt:18]: Let's keep it for now.

**Review by gino-m — APPROVED** [review]

**Pull request merged**